pkg/apis/v1alpha1/vpc_types.go

```diff
@@ -0,0 +1,138 @@
+/* Copyright � 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+   package v1alpha1
+
+   import (
+	   corev1 "k8s.io/api/core/v1"
+	   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+   )
+   
+   type VPCStatusCondition string
+   
+   const (
+	   VPCReady    VPCStatusCondition = "Ready"
+	   VPCNotReady VPCStatusCondition = "NotReady"
+   )
+   
+   //+kubebuilder:object:root=true
+   //+kubebuilder:subresource:status
+   //+kubebuilder:storageversion
+   
+   // VPC is the Schema for the VPC API
+   type VPC struct {
+	   metav1.TypeMeta   `json:",inline"`
+	   metav1.ObjectMeta `json:"metadata,omitempty"`
+   
+	   Spec   VPCSpec   `json:"spec,omitempty"`
+	   Status VPCStatus `json:"status,omitempty"`
+   }
+   
+   //+kubebuilder:object:root=true
+   
+   // VPCList contains a list of VPC
+   type VPCList struct {
+	   metav1.TypeMeta `json:",inline"`
+	   metav1.ListMeta `json:"metadata,omitempty"`
+	   Items           []VPC `json:"items"`
+   }
+   
+   // VPCSpec defines VPC configuration
+   type VPCSpec struct {
+	   // The edge cluster on which the networking elements for the Org will be created.
+	   EdgeClusterPaths []string `json:"edgeClusterPaths"`
+	   // This is the policy resource path of the Tier0 or VRF.
+	   // This must be one of Tier0s/VRFs defined at project level.
+	   // +optional
+	   GatewayPath string `json:"gatewayPath"`
+	   // Service Gateway serves as the default gateway for VPC.
+	   // +optional
+	   ServiceGateway ServiceGatewayItem `json:"serviceGateway"`
+	   // This is policy resource path of pre-created public ip block which is created by org-admin at
+	   // org level under org-default-project.
```

Comment:
org -> Org

Reply:
I feel we need not expose too many NSX-T details in CRD definition, so maybe remove "at org level under org-default-project". Just say "which is created by the Org Admin".